test/project/main.gd

```diff
@@ -3,95 +3,165 @@ extends Node
 var gdn
 
 func _ready():
-    print(" -- Rust gdnative test suite:")
-    gdn = GDNative.new()
-    var status = false;
+	print(" -- Rust gdnative test suite:")
+	_timeout()
 
-    gdn.library = load("res://gdnative.gdnlib")
+	gdn = GDNative.new()
+	var status = false;
 
-    if gdn.initialize():
-        status = gdn.call_native("standard_varcall", "run_tests", [])
+	gdn.library = load("res://gdnative.gdnlib")
 
-        status = status && _test_argument_passing_sanity()
-        status = status && _test_optional_args()
+	if gdn.initialize():
+		status = gdn.call_native("standard_varcall", "run_tests", [])
 
-        gdn.terminate()
-    else:
-        print(" -- Could not load the gdnative library.")
+		status = status && _test_argument_passing_sanity()
+		status = status && _test_optional_args()
+		status = status && yield(_test_async_resume(), "completed")
 
-    if status:
-        print(" -- Test run completed successfully.")
-    else:
-        print(" -- Test run completed with errors.")
-        OS.exit_code = 1
+		# Godot needs another frame to dispose the executor driver node. Otherwise the process
+		# aborts due to `_process` being called after `terminate` (`get_api` fail, not UB).
```

Comment:
Could this be an issue outside of tests? IE: A game shutting down?

Reply:
No. The scene tree will be completely freed before any libraries are terminated in a real game. We only have this problem in the test because we're calling `terminate` manually without using the resource system.